presto-docs/src/main/sphinx/installation/jmx.rst

```diff
@@ -0,0 +1,61 @@
+===================
+Monitoring with JMX
+===================
+
+Presto exposes a large number of different metrics via the Java Management
+Extensions (JMX). JConsole, supplied with the JDK, VisualVM, and many other
+tools can be used to access the metrics in a client application. Many monitoring
+solutions support JMX.
+
+You can configure the ports used by JMX in ``jvm.config`` as described in
+:doc:`/installation/deployment`.
+
+Alternatively you can use the :doc:`/connector/jmx` and query the metrics using
+SQL.
+
+Many of these JMX metrics are a complex metric object such as a ``CounterStat``
+that has a collection of related metrics. For example, ``InputPositions`` has
+``InputPositions.TotalCount``, ``InputPositions.OneMinute.Count``, and so on.
+
+The metrics can be grouped into a number of topics and the following sections
+describe a subset of all available metrics.
+
+JVM
+---
+
+* Heap size: ``java.lang:type=Memory:HeapMemoryUsage.used``
+* Thread count: ``java.lang:type=Threading:ThreadCount``
+
+Presto Cluster and Nodes
+------------------------
+
+* Active nodes ``presto.failureDetector:name=HeartbeatFailureDetector:ActiveCount``
+* Free memory (general pool): ``presto.memory:type=ClusterMemoryPool:name=general:FreeDistributedBytes``
+* Blocked Nodes: ``presto.memory:type=ClusterMemoryPool:name=general:BlockedNodes``
+* Number of leaked queries: ``presto.memory:name=ClusterMemoryManager:NumberOfLeakedQueries``
+* Cumulative count of killed queries since Presto start: ``presto.memory:name=ClusterMemoryManager:QueriesKilledDueToOutOfMemory``
+
+Presto Queries
+--------------
+
+* Active queries ``presto.execution:name=QueryManager:RunningQueries``
+* Failed queries (all): ``presto.execution:name=QueryManager:FailedQueries.FiveMinute.Count``
+* Failed queries (internal): ``presto.execution:name=QueryManager:InternalFailures.FiveMinute.Count``
+* Failed queries (external): ``presto.execution:name=QueryManager:ExternalFailures.FiveMinute.Count``
+* Failed queries (user): ``presto.execution:name=QueryManager:UserErrorFailures.FiveMinute.Count``
+* Assigned queries: ``presto.memory:type=ClusterMemoryPool:name=general:AssignedQueries``
+* Queries started: ``presto.execution:name=QueryManager:StartedQueries.FiveMinute.Count``
+* Execution latency (P50): ``presto.execution:name=QueryManager:ExecutionTime.FiveMinutes.P50``
+* Input data rate (P90): ``presto.execution:name=QueryManager:WallInputBytesRate.FiveMinutes.P90``
+
+Presto Tasks
+------------
+
+* Input data bytes: ``presto.execution:name=TaskManager:InputDataSize.FiveMinute.Count``
+* Input rows: ``presto.execution:name=TaskManager:InputPositions.FiveMinute.Count``
+
+Hive Connector
+--------------
+
+* HDFS NameNode latency (P99): ``presto.plugin.hive:name=hive,type=NamenodeStats:ListLocatedStatus.Time.OneMinute.P99``
```

Comment:
this is only one of NamenodeStats, right? if so, shouldn't be labeled "HDFS NameNode latency"

Reply:
Not sure .. to be honest I assume not. There are probably a whole lot more .. keep in mind all these listed properties are just examples for probably a few hundred properties overall that we might want to document over time...